lib/pipeline.js

```diff
@@ -1296,16 +1313,25 @@ class PipelineModel extends BaseModel {
         const allEvents = await getAllRecords.call(this,
             'getEvents', config.aggregateInterval, options, [[]]);
 
-        return Promise.all(allEvents.map(async (sameDateEvents) => {
-            const { totalQueuedTime, totalImagePullTime, totalDuration, emptyEvent }
-                = await sumAggregatedEventMetrics(sameDateEvents);
-            const length = sameDateEvents.length - emptyEvent;
+        return Promise.all(allEvents.map(async (sameIntervalEvents) => {
+            const length = sameIntervalEvents.length;
+            const { totalQueuedTime, totalImagePullTime, totalDuration, emptyCount }
+                = await sumAggregatedEventMetrics(sameIntervalEvents);
+            const duration = length > emptyCount.duration
```

Comment:
wondering if this logic would be better in a helper function, but we can refactor that at a different time